TombLib/TombLib/Catalogs/TEN Node Catalogs/Particles.lua

```diff
@@ -0,0 +1,114 @@
+--!Section "Particles"
+--!Conditional "False"
+--!Description "Emit a lightning arc between two points in 3D space"
+--!Arguments "Newline, Moveables, 50, Source position", "Moveables, 50, Destination position"
+--!Arguments "NewLine, Color, 100, Color of Lightning Effect"
+--!Arguments "Newline, Number, 25, [ 0 | 4.2 | 1 ], Lifetime in seconds." , "Number, 25, [ 1 | 255 | 0 ], Effect strength", "Number, 25, [ 1 | 127 | 0 ], Beam width", "Number, 25, [ 1 | 127 | 0 ], Detail level\n 1 is 1 segment between the two points for example."
+--!Arguments "Newline, Boolean, 50, Toggle smooth effect.", "Boolean, 50, Toggle end drift."
+--!Arguments "NewLine, Boolean, 50, Add source light" , "Boolean, 50, Add destination light"
+
+LevelFuncs.Engine.Node.LightningArc = function(source, dest, color, lifetime, amplitude, beamWidth, detail, smooth, endDrift, sourcelight, destlight)
+	
+	local randomiserX = math.random(-64,64)
+	local randomiserZ = math.random(-256,256)
+	
+    local entity = TEN.Objects.GetMoveableByName(source)
+	
+    local startingpoint = entity:GetPosition()
+		startingpoint.x = startingpoint.x - randomiserX
+		startingpoint.y = startingpoint.y - 64
+		startingpoint.z = startingpoint.z - randomiserZ 	
+		
+	local endingpoint = TEN.Objects.GetMoveableByName(dest):GetPosition()
+		endingpoint.x = endingpoint.x - randomiserX
+		endingpoint.y = endingpoint.y - 64
+		endingpoint.z = endingpoint.z - randomiserZ 
+		
+	local beamrandom = math.random((beamWidth-10),(beamWidth+10))
+	local amprandom =  math.random((amplitude-10),(amplitude+10))
+	
+	local sourceLightCheck 
+		if sourcelight == true
+		then
+			TEN.Effects.EmitLight(startingpoint, color, math.random(1,10))
+		end
+
+	local destLightCheck 
+		if destlight == true
+		then 
+			TEN.Effects.EmitLight(endingpoint, color, math.random(1,10))
+		end
+	TEN.Effects.EmitLightningArc(startingpoint, endingpoint, color, lifetime, amprandom, beamrandom, detail, smooth, endDrift)
+	
+
+end
+
+--!Name "Emit shockwave"
+--!Section "Particles"
+--!Description "Emit a shockwave effect."
+--!Arguments "NewLine, Moveables, 70, Shockwave position" "Numerical, 30, [0 | 100], Mesh number/n(optional)"
+--!Arguments "NewLine, Number, 50, [ 1 | 10400 | 0 ], Inner radius" "Number, 50, [ 1| 10400 |0 ], Outer radius."
+--!Arguments "NewLine, Color, 100, Color of shockwave effect"
+--!Arguments "NewLine, Numerical, 33, [ 0 | 8.5 | 1 ], Lifetime of effect (in seconds)" , "Number, 33, [ 0 | 500 | 0 ], Speed" , "Number, 33, [ -360 | 360 |0 ], X axis rotation"
+--!Arguments "NewLine, Boolean, 50, Damage" "Boolean, 50, Randomise spawn point"
+
+	LevelFuncs.Engine.Node.Shockwave = function(pos, meshnum, innerRadius, outerRadius, color, lifetime, speed, angle, damage, randomise)
+			
+		local randomiser = math.random(1,14)
+		local radiusInVar =  innerRadius + math.random(1,3)
+		local radiusOutVar = outerRadius + math.random(1,3)
+		local randomMesh = math.random(0,14)
+		
+			local randomCheck
```

Comment:
What is this? What is `randomCheck`? Where it is used?

Reply:
Can I just check if this code:

```lua

    local randomiser = math.random( 1, 14)
    local radiusInVar =  innerRadius + math.random( 1, 3)
    local radiusOutVar = outerRadius + math.random( 1, 3)
    local randomMesh = math.random( 0, 14)

    if (randomSpawn == true) then
            local origin = TEN.Objects.GetMoveableByName(pos):GetJointPosition(randomMesh)
            TEN.Effects.EmitShockwave(origin, radiusInVar, radiusOutVar, color, lifetime, speed, angle, damage)
        else
            local origin = TEN.Objects.GetMoveableByName(pos):GetJointPosition(meshnum)
            TEN.Effects.EmitShockwave(origin, radiusInnerVar, radiusOutVar, color, lifetime, speed, angle, damage)
        end
    end

```

is correct? I removed the ```lua

local randomCheck

``` 

section but Im not sure if its correct?

Reply:
`local randomCheck` makes no sense. It looks like you are trying to initialize a new variable. Then you immediately start an if statement on the next line. What you were trying to achieve with that?

Regarding your code above: this block by itself is not correct, because you again have two `end` statements at the end of if-then-else block. It should go as:

```if (foo == true) then
    blah
else
    bar
end```